# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/DMABufSurface.cpp
# Commit: a04ffaf75943
# Full Hash: a04ffaf75943efc2fa4472c36ba8ca9b7b452195
# Author: stransky <stransky@redhat.com>
# Date: 2025-05-08 04:32:48
# Description:
#   Bug 1964924 [Linux] In case of missing DRMFormat create DMABufSurfaceGBM with stored modifiers r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D248191
# ==============================================================================

diff -r d2f1231bdfc6 -r a04ffaf75943 widget/gtk/DMABufSurface.cpp
--- a/widget/gtk/DMABufSurface.cpp	Wed May 07 09:47:59 2025 +0000
+++ b/widget/gtk/DMABufSurface.cpp	Wed May 07 10:00:55 2025 +0000
@@ -1540,8 +1540,8 @@
 
   MOZ_DIAGNOSTIC_ASSERT(mGbmBufferObject[aPlane] == nullptr);
 
-  if (aFormat->UseModifiers()) {
-    LOGDMABUF("    Creating with modifiers\n");
+  if (aFormat && aFormat->UseModifiers()) {
+    LOGDMABUF("    Creating with modifiers from DRMFormat");
     uint32_t modifiersNum = 0;
     const uint64_t* modifiers = aFormat->GetModifiers(modifiersNum);
     mGbmBufferObject[aPlane] = GbmLib::CreateWithModifiers2(
@@ -1550,6 +1550,12 @@
     if (mGbmBufferObject[aPlane]) {
       mBufferModifiers[aPlane] = GbmLib::GetModifier(mGbmBufferObject[aPlane]);
     }
+  } else if (mBufferModifiers[aPlane] != DRM_FORMAT_MOD_INVALID) {
+    LOGDMABUF(
+        "    Creating with modifiers from DMABufSurface mBufferModifiers");
+    mGbmBufferObject[aPlane] = GbmLib::CreateWithModifiers2(
+        GetDMABufDevice()->GetGbmDevice(), mWidth[aPlane], mHeight[aPlane],
+        mDrmFormats[aPlane], mBufferModifiers + aPlane, 1, mGbmBufferFlags);
   }
   if (!mGbmBufferObject[aPlane]) {
     LOGDMABUF("    Creating without modifiers");
